frontend/src/features/properties/map/research/ViewSelector.tsx

```diff
@@ -38,18 +52,29 @@ const ViewSelector: React.FunctionComponent<IViewSelectorProps> = props => {
     const researchFileProperty = properties[selectedPropertyIndex];
     researchFileProperty.researchFile = props.researchFile;
     if (props.isEditMode) {
-      return (
-        <UpdatePropertyView
-          researchFileProperty={researchFileProperty}
-          setFormikRef={props.setFormikRef}
-          onSuccess={props.onSuccess}
-        />
-      );
+      if (props.editKey === FormKeys.propertyDetails) {
+        return (
+          <UpdatePropertyDetailsContainer
+            id={researchFileProperty.property?.id as number}
+            onSuccess={props.onSuccess}
+            setFormikRef={props.setFormikRef}
+          />
+        );
+      } else {
+        return (
+          <UpdatePropertyView
+            researchFileProperty={researchFileProperty}
+            setFormikRef={props.setFormikRef}
+            onSuccess={props.onSuccess}
+          />
+        );
+      }
```

Comment:
The logic on ViewSelector is becoming hard to follow with multiple nested if statements. This would be a good candidate for refactoring but will not do as part of the bug fix - should be addressed separately (in my opinion)